### PR TITLE
Implement PSR7 header methods

### PR DESCRIPTION
**Pull request by markstory**
Implement the header related method required for the PSR7 ServerRequestInterface. 

I've chosen to make the new methods read from the existing data as with the other changes I've done so far. This should hopefully make the new methods non-destructive to the existing ones.

Refs #9325

### CODE REVIEW

File: src/Network/Request.php [hunk 1]
```diff
@@ -367,8 +367,8 @@ protected function _processPost($data)
             $data = $this->input();
             parse_str($data, $data);
         }
-        if ($this->env('HTTP_X_HTTP_METHOD_OVERRIDE')) {
-            $data['_method'] = $this->env('HTTP_X_HTTP_METHOD_OVERRIDE');
+        if ($this->hasHeader('X-Http-Method-Override')) {
+            $data['_method'] = $this->getHeaderLine('X-Http-Method_Override');
```
Comment (lorenzo) [inline src/Network/Request.php]: Why does it have an underscore here?

Comment (lorenzo) [inline src/Network/Request.php]: I meant, there is an underscore instead of a hyphen in the header name.

File: src/Network/Request.php [hunk 2]
```diff
@@ -944,22 +944,175 @@ public function here($base = true)
...
+     * Check if a header exists in the request.
+     *
+     * @param string $name The header you want to get (case-insensitive)
+     * @return bool Whether or not the header is defined.
+     */
+    public function hasHeader($name)
+    {
+        $name = $this->normalizeHeaderName($name);
+
+        return array_key_exists($name, $this->_environment);
```
Comment (lorenzo) [inline src/Network/Request.php]: I think isset is more appropriate here. At least that is what the get function is using

**Review by lorenzo — CHANGES_REQUESTED** [review]

File: src/Network/Request.php (hunk 1 above)
Comment (markstory) [inline src/Network/Request.php]: Because I'm a muppet.

File: src/Network/Request.php (hunk 2 above)
Comment (markstory) [inline src/Network/Request.php]: Fixed.

**Comment by lorenzo** [comment]
Great job @markstory

**Comment by markstory** [comment]
Uri methods are next!

**Pull request merged**